### ISSUE TITLE
Confusing custom label configuration

### ISSUE THREAD

**Issue by keithhand** [kubecost/cost-analyzer-helm-chart#1665]
**Describe the bug**
When using custom aggregations (Product, Environment, Department), if you refer to a label containing a character Prometheus doesn't like ([more info](https://guide.kubecost.com/hc/en-us/articles/4407595916823-Allocation-API#allocation-schema-version-1-76)), the aggregation looks like it works, however, drilling down will fail. You can get around this by manually converting the label to use `_` characters.

**To Reproduce**
1. Configure Kubecost with the custom owner/team labels
```yaml
## Test label
global:
  additionalLabels: 
    kubecost.com/test-label: test-four # product

## Label configs
kubecostProductConfigs:
  labelMappingConfigs:
    enabled: true
    owner_label: "kubecost.com/test-label"
    team_label: "kubecost_com_test_label" # workaround
```
2. Check allocation report based on owner aggregation

![not-working-owner](https://user-images.githubusercontent.com/43491549/186783120-516f9d96-62a3-466b-877a-25a6fef341a4.gif)

**Expected behavior**
![working-team](https://user-images.githubusercontent.com/43491549/186783647-5e772a1a-d533-4a4e-9ba9-970c8a402653.gif)

gz#2461<br><br><i>(related to <a href='https://kubecost.zendesk.com/agent/tickets/2461'>Zendesk ticket #2461</a>)</i>

**Comment by dwbrown2** [comment]
Really interesting. @nealormsbee have any intuition around this one? I was going to say this is a Prom limitation but doesn't seem like it given the initial page loads.

**Comment by nealormsbee** [comment]
I can check this out. My guess is that the initial query is converting the label to be prom-friendly, but the drilldown logic isn't.

**Comment by nikovacevic** [comment]
This is another great example of inconsistencies between aggregation and filtering, where we've added a feature to one and not another. Good news: this should be a quick fix.

## Aggregate
Note the calls to `labelConfig.Sanitize` in the working examples below:

https://github.com/opencost/opencost/blob/f501e82f7fe0a0f26c7a031bd8b28421d7ddb4d8/pkg/kubecost/allocationprops.go#L299
```go
case strings.HasPrefix(agg, "label:"):
	labels := p.Labels
	if labels == nil {
		names = append(names, UnallocatedSuffix)
	} else {
		labelName := labelConfig.Sanitize(strings.TrimPrefix(agg, "label:"))
		if labelValue, ok := labels[labelName]; ok {
			names = append(names, fmt.Sprintf("%s=%s", labelName, labelValue))
		} else {
			names = append(names, UnallocatedSuffix)
		}
	}
```
https://github.com/opencost/opencost/blob/f501e82f7fe0a0f26c7a031bd8b28421d7ddb4d8/pkg/kubecost/allocationprops.go#L318
```go
case agg == AllocationDepartmentProp:
	labels := p.Labels
	annotations := p.Annotations
	if labels == nil && annotations == nil {
		names = append(names, UnallocatedSuffix)
	} else {
		labelNames := strings.Split(labelConfig.DepartmentLabel, ",")
		for _, labelName := range labelNames {
			labelName = labelConfig.Sanitize(labelName)
			if labelValue, ok := labels[labelName]; ok {
				names = append(names, labelValue)
			} else if annotationValue, ok := annotations[labelName]; ok {
				names = append(names, annotationValue)
			} else {
				names = append(names, UnallocatedSuffix)
			}
		}
	}
```

## Filtering
In allocation filtering, we're not doing the same sanitization -- and in fact, I don't see where we're converting label aliases. (It's likely happening somewhere, but seemingly not here.)

https://github.com/opencost/opencost/blob/develop/pkg/kubecost/allocationfilter.go#L313

cc @michaelmdresser -- maybe we can discuss this one?

**Comment by michaelmdresser** [comment]
v1 filters are now parsed using https://github.com/opencost/opencost/blob/f501e82f7fe0a0f26c7a031bd8b28421d7ddb4d8/pkg/util/allocationfilterutil/queryfilters.go#L43, which appears to be doing sanitization for label and annotation filters: https://github.com/opencost/opencost/blob/f501e82f7fe0a0f26c7a031bd8b28421d7ddb4d8/pkg/util/allocationfilterutil/queryfilters.go#L325. If we aren't experiencing that behavior, we should double check the code path and make sure its using this parse logic.

I don't think I implemented label-/annotation-mapping in v2 filters, partially due to time constraints and partially due to how weird it feels to make the mapping belong to the parser, rather than a higher query level (like the frontend!). With that said, this is definitely a feature gap we need to fill. For what its worth, the frontend isn't yet using v2 filters, see https://github.com/kubecost/cost-analyzer-frontend/issues/925.

I'm happy to take some time to discuss either/both of these.

**Comment by michaelmdresser** [comment]
Niko and I identified the source of this error -- a lack of sanitization in one of our v1 filter parsing code paths. Fix PR is up: https://github.com/opencost/opencost/pull/1392.

**Comment by michaelmdresser** [comment]
Now that the fix is merged, this will be fixed in 1.98. Currently in discussion to determine if this will be cherry picked into the v1.97 release to get it out earlier.

**Comment by michaelmdresser** [comment]
Update: this issue should be fixed in v1.97 of Kubecost (the upcoming release).
